src/tools/run-make-support/src/rustc.rs

```diff
@@ -10,7 +10,13 @@ pub fn rustc() -> Rustc {
     Rustc::new()
 }
 
-/// Construct a new `rustc` aux-build invocation.
+/// Construct a plain `rustc` invocation with no flags set.
+#[track_caller]
+pub fn bare_rustc() -> Rustc {
+    Rustc::bare()
+}
+
+// Construct a new `rustc` aux-build invocation.
```

Comment:
Suggestion: this can be a doc comment, maybe also explain that we set `HOST_RPATH`/`TARGET_RPATH` (it might be doing some other things, I don't recall exactly). If `aux_build()` *isn't* used in tests (they were introduced initially because Makefiles used `AUX_BUILD` IIRC) but now we can just use `rustc()` vs `bare_rustc()` which seems clearer, then I recommend nuking them.

Reply:
> maybe also explain that we set HOST_RPATH/TARGET_RPATH
- You mean, in compiletest? This part? Alright, I gave it a try.
```rs
.env("HOST_RPATH_DIR", cwd.join(&self.config.compile_lib_path))
.env("TARGET_RPATH_DIR", cwd.join(&self.config.run_lib_path))
```

> which seems clearer, then I recommend nuking them.

What is "them" referring to here? The `aux_build()`? Some `rmake.rs` do use it, but it seems reasonably easy to change it.

Reply:
I was referring to `set_host_rpath` in `setup_common`, sorry for not being super clear.

Please disregard the "nuking them" comment, as `aux_build` looked similar but it defaulted to `--crate-type=lib` when compared to `bare_rustc` so they're not interchangeable.